metrics-prometheus/src/main/java/com/alibaba/metrics/prometheus/DubboMetricsExports.java

```diff
@@ -0,0 +1,187 @@
+                metricFamilySamples.add(fromClusterHistogram(entry.getKey(), entry.getValue()));
+            }
+            for (Map.Entry<MetricName, Compass> entry : metricRegistry.getCompasses().entrySet()) {
+                metricFamilySamples.add(fromCompass(entry.getKey(), entry.getValue()));
+            }
+            for (Map.Entry<MetricName, FastCompass> entry : metricRegistry.getFastCompasses().entrySet()) {
+                metricFamilySamples.add(fromFastCompass(entry.getKey(), entry.getValue()));
+            }
+        }
+        return metricFamilySamples;
+    }
+
+    public MetricFamilySamples fromCounter(MetricName metricName, Counter counter) {
+        MetricFamilySamples.Sample sample = sampleBuilder.createSample(metricName, "", new ArrayList<String>(), new ArrayList<String>(), new Long(counter.getCount()).doubleValue());
+        return new MetricFamilySamples(sample.name, Type.COUNTER, getHelpMessage(metricName.getKey(), counter), Arrays.asList(sample));
+    }
+
+    public MetricFamilySamples fromGauge(MetricName metricName, Gauge gauge) {
+        Object o = gauge.getValue();
+        double value;
+        if (o instanceof Number) {
+            value = ((Number) o).doubleValue();
+        } else if (o instanceof Boolean) {
+            value = ((Boolean) o) ? 1: 0;
+        } else {
+            return null;
+        }
+        MetricFamilySamples.Sample sample = sampleBuilder.createSample(metricName, "", new ArrayList<String>(), new ArrayList<String>(), value);
+        return new MetricFamilySamples(sample.name, Type.GAUGE, getHelpMessage(metricName.getKey(), gauge), Arrays.asList
+                (sample));
+    }
+
+    public MetricFamilySamples fromTimer(MetricName metricName, Timer timer) {
+        return fromSnapshotAndCount(metricName, timer.getSnapshot(), timer.getCount(),
+                1.0D / TimeUnit.SECONDS.toNanos(1L), getHelpMessage(metricName.getKey(), timer));
+    }
+
+    public MetricFamilySamples fromMeter(MetricName metricName, Meter meter) {
+        MetricFamilySamples.Sample sample = sampleBuilder.createSample(metricName, "_total", new ArrayList<String>(), new ArrayList<String>(), meter.getCount());
+        return new MetricFamilySamples(sample.name, Type.COUNTER, getHelpMessage(metricName.getKey(), meter), Arrays.asList
+                (sample));
+    }
+
+    public MetricFamilySamples fromHistogram(MetricName metricName, Histogram histogram) {
+        return fromSnapshotAndCount(metricName, histogram.getSnapshot(), histogram.getCount(), 1.0,
+                getHelpMessage(metricName.getKey(), histogram));
+    }
+
+    public MetricFamilySamples fromClusterHistogram(MetricName metricName, ClusterHistogram clusterHistogram) {
+        List<MetricFamilySamples.Sample> samples = new ArrayList<MetricFamilySamples.Sample>();
+        Map<Long, Map<Long, Long>> buckets = clusterHistogram.getBucketValues(0L);
```

Comment:
should pass the current time stamp rather than `0L`, or it will retrieve all the buckets.